api/base/views.py

```diff
@@ -6,57 +6,196 @@
+
+    + `relationships`
+
+    Relationships are urls to other entities or collections that have a relationship to the entity. For example, the
+    node entity provides a `contributors` relationship that points to the endpoint to retreive all contributors to that
+    node.  It is reccommended to use these links rather than to id-filter of general collection endpoints.  They'll be
+    faster, easier, and less error-prone.
+
+    + `links`
+
+    Links are urls to alternative representations of the entity or actions that may be performed on the entity.  Most
+    entities will provide a `self` link that is the canonical endpoint for the entity where update and delete requests
+    should be sent.  In-depth documentation of actions is available by navigating to the `self` link in the Browsable
+    API.  Most entities will also provide an `html` link that directs to the entity's page on the [OSF](http://osf.io/).
+
+    ###Collections
+
+    Collection endpoints return a list of entities and an additional data structure with pagination links, such as
+    "next", "prev", "first", and "last". The OSF API limits all collection responses to a maximum of 10 entities.  The
+    response object has two keys:
+
+    + `data`
+
+    `data` is an array of entities that match the query.  Each entity in the array is the same representation that is
+    returned from that entity's `self` link, meaning that refetching the entity is unnecessary.
+
+    + `links`
+
+    `links` contains pagination information, including links to the previous, next, first, and last pages of results.
+    The meta key contains the total number of entities available, as well as the current number of results displayed per
+    page.  If there are only enough results to fill one page, the `first`, `last`, `prev`, and `next` values will be
+    null.
+
+    ###PUT vs. PATCH
+
+    For most endpoints that support updates via PUT requests, we also allow PATCH updates.  The only difference is that
+    PUT requests require all mandatory attributes to be set, even if their value is unchanged.  PATCH requests may omit
+    mandatory attributes, whose value will be unchaged.
+
+    ###Attribute Validation
+
+    Endpoints that allow creation or modification of entities generally limit updates to certain attributes of the
+    entity.  If you attempt to set an attribute that does not permit updates (such as a `date_created` timestamp), the
+    API will silently ignore that attribute.  This will not affect the response from the API: if the request would have
+    succeeded without the updated attribute, it will still report as successful.  Likewise, if the request would have
+    failed without the attribute update, the API will still report a failure.
 
+    Typoed or non-existant attributes will behave the same as non-updatable attributes and be silently
+    ignored. If a request is not working the way you expect, make sure to double check your spelling.
```

Comment:
I'm having Larry add an issue for improving this. I think successful requests will report all unused parameters in a meta field so people can see what didn't go into their requests. This would be a not-breaking method of making this work, presuming it's possible.